transactions/brc20.ts

```diff
@@ -37,3 +60,306 @@ export const createBrc20TransferOrder = async (
     feesResponse,
   };
 };
+
+export const brc20MintEstimateFees = async (estimateProps: EstimateProps) => {
+  const { addressUtxos, tick, amount, revealAddress, feeRate } = estimateProps;
+  const dummyAddress = 'bc1pgkwmp9u9nel8c36a2t7jwkpq0hmlhmm8gm00kpdxdy864ew2l6zqw2l6vh';
+
+  const { chainFee: revealChainFee, serviceFee: revealServiceFee } = await xverseInscribeApi.getBrc20MintFees(
+    tick,
+    amount,
+    revealAddress,
+    feeRate,
+    FINAL_SATS_VALUE,
+  );
+
+  const commitValue = new BigNumber(FINAL_SATS_VALUE).plus(revealChainFee).plus(revealServiceFee);
+
+  const bestUtxoData = selectUtxosForSend(
+    dummyAddress,
+    [{ address: revealAddress, amountSats: new BigNumber(commitValue) }],
+    addressUtxos,
+    feeRate,
+  );
+
+  if (!bestUtxoData) {
+    throw new Error('Not enough funds at selected fee rate');
+  }
+
+  const commitChainFees = bestUtxoData.fee;
+
+  return {
+    commitValue: commitValue.plus(commitChainFees).toNumber(),
+    valueBreakdown: {
+      commitChainFee: commitChainFees,
+      revealChainFee,
+      revealServiceFee,
+      inscriptionValue: FINAL_SATS_VALUE,
+    },
+  };
+};
+
+export async function brc20MintExecute(executeProps: ExecuteProps): Promise<string> {
+  const { seedPhrase, accountIndex, addressUtxos, tick, amount, revealAddress, changeAddress, feeRate, network } =
+    executeProps;
+
+  const privateKey = await getBtcPrivateKey({
+    seedPhrase,
+    index: BigInt(accountIndex),
+    network: 'Mainnet',
+  });
+
+  const { commitAddress, commitValue } = await xverseInscribeApi.createBrc20MintOrder(
+    tick,
+    amount,
+    revealAddress,
+    feeRate,
+    network,
+    FINAL_SATS_VALUE,
+  );
```

Comment:
there's no error handling on the api method or here, so there's high potential for "destructuring of undefined" error here

Reply:
That shouldn't be the case. If the call throws an error, then the error will propagate down instead of returning `undefined`. We actually handle errors in quite a few places in core and return `null | undefined`, which we should probably remove. Returning undefined is very different to a call failing.

Reply:
ok, makes sense. I had it wrong, since I just looked up Axios Response and data will always be {}, or the call will throw.
so basically we don't want to return null or undefined from any of our api call methods